### ISSUE TITLE
Impossible to change the type of a layer

### ISSUE THREAD

**Issue by doskabouter** [maputnik/editor#738]
Maputnik version 1.7.0
Latest firefox on windows

Click to select any layer, there's no dropdown anymore next to the "type" field of the layer

**Comment by pathmapper** [comment]
@doskabouter thanks for bringing this up.

Currently it's possible to change the type of a layer using the JSON Editor:

![change_layer_type](https://user-images.githubusercontent.com/20856381/95008418-0a7fdc00-061a-11eb-9fc1-1e10e90f4023.gif)

The different layer types are having different properties, so depending on the properties used it could be that switching a layer type leads to a bunch of  errors.

I think there is no dropdown for historic reasons because in the past we didn't have a way to have a partially invalid style.

But now it should be possible to add the dropdown for layer type. I would find this useful especially for cases where you've just added a layer and selected the wrong type in the `Add Layer` modal and want to switch to the wanted type.

**Comment by doskabouter** [comment]
Thanks for the response, changing it in the json is ofcourse a possibiity.
Re no dropdown: I'm reasonably sure I saw one in one of the previous versions...

**Comment by pathmapper** [comment]
> I'm reasonably sure I saw one in one of the previous versions...

@doskabouter ohh, you're right. It was available in `v1.6.1` and older releases:
https://maputnik.github.io/releases/v1.6.1/ 

@orangemug was this removed intentional?

**Comment by chau-intl** [comment]
As it does make sense to be able to change the layer type it would be nice if it could be re-introduced. 

I used to use this feature too in one of the previous versions and our users are really confused about it being missing since they like to use Maputnik because it provides a UI instead of having to write JSON code.

**Comment by orangemug** [comment]
Changing a type makes some (most??) of the properties of the previous layer type invalid. For example changing from a `circle` to a `line` makes everything except the common properties invalid in the new layer type, `circle-color` doesn't work with a `line` layer.

@chau-intl what does the user expect the app to do in this situation?

Also @chau-intl if you have time can you comment in #164 and tell us what you're up to with Maputnik?

**Comment by pathmapper** [comment]
> what does the user expect the app to do in this situation?

Maybe keep the common properties and use default values according to the style spec for the others?

**Comment by chau-intl** [comment]
@orangemug, well changing the type of course implies that a lot of settings might become irrelevant and I guess the user is okay with them being reset. I think what @pathmapper suggests above would be the expected result.

**Comment by orangemug** [comment]
@chau-intl yeah true I guess we do have an undo stack as well so they can always revert the change.

Yeah I'm happy with that as an approach.
